pkg/blockbuilder/blockbuilder.go

```diff
@@ -541,9 +598,13 @@ func (b *BlockBuilder) consumePartition(
 		b.kafkaClient.AllowRebalance()
 	}
 
+	consumerLagMetric.Set(float64(lag))
+	start := time.Now()
 	if err := builder.compactAndUpload(ctx, b.blockUploaderForUser); err != nil {
```

Comment:
I'd move this inside the `builder` (and also I would add more metrics to the builder itself). I.e. I'd love to track the builder's processing latency.

Reply:
I tried to move this metric inside builder but it is getting convoluted. Logically does not make sense to pass the topic, partition, to the builder just for the metric. I think we should keep the metric here for now.